### PR TITLE
PrivateAPI: history

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
private api

### Description
This PR adds the history route to the private api.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Review by InnayTool — APPROVED** [review]

**Review by InnayTool — CHANGES_REQUESTED** [review]
see comment above

File: src/history/history.service.ts [hunk 1]
```diff
@@ -93,6 +92,16 @@ export class HistoryService {
     return;
   }
 
+  async deleteHistory(user: User): Promise<void> {
+    const entries: HistoryEntry[] = await this.getEntriesByUser(user);
+    if (!entries) {
```
Comment (davidmehren) [inline src/history/history.service.ts:97]: When does `getEntriesByUser` return a nullish value?

File: src/history/history.service.ts [hunk 2]
```diff
@@ -93,6 +92,16 @@ export class HistoryService {
     return;
   }
 
+  async deleteHistory(user: User): Promise<void> {
```
Comment (davidmehren) [inline src/history/history.service.ts:95]: Please add JSDoc annotations for (at least new) service methods

**Review by davidmehren — CHANGES_REQUESTED** [review]
Using a route under `/history` to get the history entries for the current user is inconsistent with the public API, which uses the `/me` route for things that relate to the current user.
Why do you want to change that for the private API?

I'd just move the whole thing to `/me/history` to mirror the public API.

(I realize we already merged `/tokens` which has the same issue and should also be moved)

File: src/history/history.service.ts (hunk 2 above)
Comment (DerMolly) [reply src/history/history.service.ts:139]: added

File: src/api/private/history/history.controller.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    this.logger.setContext(HistoryController.name);
+  }
+
+  @Get()
+  async getHistory(): Promise<HistoryEntryDto[]> {
+    // ToDo: use actual user here
+    try {
+      const user = await this.userService.getUserByUsername('hardcoded');
+      const foundEntries = await this.historyService.getEntriesByUser(user);
+      return await Promise.all(
```
[the remark this answers is not in the document]
Comment (DerMolly) [reply src/api/private/history/history.controller.ts:45]: we could probably remove that

[the remark this answers is not in the document]
Comment (DerMolly) [reply src/api/private/history/history.controller.ts]: fixed

File: src/history/history.service.ts (hunk 1 above)
Comment (DerMolly) [reply src/history/history.service.ts]: Never, it's just a habit because `findOne` returns `undefined`.

**Review by InnayTool — APPROVED** [review]

**Comment by DerMolly** [comment]
> I'd just move the whole thing to /me/history to mirror the public API.

I just did that

**Comment by InnayTool** [comment]
Maybe we should move the `history.controller.ts` to `src/api/private/history/history.controller.ts`.
I think it makes the structure clearer.

**Comment by DerMolly** [comment]
> Maybe we should move the `history.controller.ts` to `src/api/private/history/history.controller.ts`.
> I think it makes the structure clearer.

You probably mean `src/api/private/me/history/history.controller.ts`, don't you?

**Comment by InnayTool** [comment]
I've updated the statement. I copied the wrong path. But maybe your suggestion is more consistent even if I don't like the very long path and the extra sub-directory.

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**